### DISCUSSION TITLE
Has there been change in SSL/TLS backends with PowerShell 7 ?

### DISCUSSION THREAD

**Discussion by bplessis** [PowerShell/PowerShell#16333] [Q&A]
Hi,

I have been experiencing issues requesting websites like api.cloudflare.com or ipinfo.io with Invoke-WebRequest/RestMethod.
My investigation point to some TLS 1.3 compatibility issue and some (yet unfound) gpo/registry settings having enabling TLS1.3 support on the system.

However when trying to overcome this setting i found that i wasn't able to in powershell 7.1 using the 'good old' [System.Net.ServicePointManager]::SecurityProtocol variable. Is it to be expected ? I can't find documentation on the change

Exemple with PowerShell 5.1:
```
PS C:\Users\####> [System.Net.ServicePointManager]::SecurityProtocol                                          SystemDefault
PS C:\Users\####>  Invoke-WebRequest -Method GET -Uri "https://ipinfo.io/json"
Invoke-WebRequest : La demande a été abandonnée : Impossible de créer un canal sécurisé SSL/TLS.
Au caractère Ligne:1 : 2
+  Invoke-WebRequest -Method GET -Uri "https://ipinfo.io/json"
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation : (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebEx
   ception
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand

PS C:\Users\####> [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::Tls12
PS C:\Users\####>  Invoke-WebRequest -Method GET -Uri "https://ipinfo.io/json"

StatusCode        : 200
StatusDescription : OK
```

But with PowerShell 7.1.5:
```
PS C:\Users\####> [System.Net.ServicePointManager]::SecurityProtocol
SystemDefault
PS C:\Users\####>  Invoke-WebRequest -Method GET -Uri "https://ipinfo.io/json"
Invoke-WebRequest: The read operation failed, see inner exception.
PS C:\Users\####> [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::Tls12
PS C:\Users\####>  Invoke-WebRequest -Method GET -Uri "https://ipinfo.io/json"
Invoke-WebRequest: The read operation failed, see inner exception.
```

Am i missing something ?
Also any pointer to search for why TLS 1.3 is active on the client side on this particular computer by the way ???

**Comment by bplessis** [comment]
Also i have the same opposite reaction when working for a computer with TLS 1.3 not enabled: enabling it with powershell 5.1 trigger the connection issue, while trying to enable it in powershell 7.1 doesn't do a thing

**Comment by iSazonov** [comment]
ServicePointManager is not used any more and present in .Net for backward compatibility.

We added TLS13 in PowerShell 7.2. I'd suggest you to try latest 7.2 Preview/RC (release is expected in November) .
```powershell
Invoke-WebRequest -Method GET -Uri "https://ipinfo.io/json" -SslProtocol Tls13
```

Really PowerShell does nothing with Tls - it exposes .Net API and .Net delegates Tls support to underlying OS.

**Reply by bplessis** [reply]
Well actually my need was more to disable Tls13 because the protocol implemented by schannel and used by powershell at least up to 7.1.5 is not compatible with 'non-microsoft' server. (And similarly IIS with microsoft TLS.13 enabled won't work with anything but iexplore and maybe edge)

**Reply by iSazonov** [reply]
PowerShell 7 uses OS default. So if you need to disable Tls13 you should do this on OS level.
